kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/BatchesResourceSuite.scala

```diff
@@ -437,10 +437,12 @@ class BatchesResourceSuite extends KyuubiFunSuite with RestFrontendTestHelper wi
     assert(session2.createTime === batchMetadata2.createTime)
 
     eventually(timeout(5.seconds)) {
-      assert(session1.batchJobSubmissionOp.getStatus.state === OperationState.RUNNING)
+      assert(session1.batchJobSubmissionOp.getStatus.state === OperationState.RUNNING ||
+        session1.batchJobSubmissionOp.getStatus.state === OperationState.FINISHED)
       assert(session1.batchJobSubmissionOp.builder.processLaunched)
 
-      assert(session2.batchJobSubmissionOp.getStatus.state === OperationState.RUNNING)
+      assert(session2.batchJobSubmissionOp.getStatus.state === OperationState.RUNNING ||
+        session2.batchJobSubmissionOp.getStatus.state === OperationState.FINISHED)
```

Comment:
why these happens in a refactoring

Reply:
it's a flaky test before this pr, see https://github.com/apache/incubator-kyuubi/pull/3159